packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -502,6 +505,30 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     };
     _updateActions();
     _updateOverlay();
+
+    WidgetsBinding.instance.addObserver(this);
+    WidgetsBinding.instance.addPostFrameCallback((_) => _getFieldWidth());
+  }
+
+  bool _isOffstage = true;
+  double? _fieldWidth;
+  final GlobalKey _textFieldKey = GlobalKey();
+
+  void _getFieldWidth() {
+    assert(_textFieldKey.currentContext != null, "The fieldKey must be assigned to a TextFormField");
+    if (_textFieldKey.currentContext != null) {
+      setState(() {
+        _fieldWidth = _textFieldKey.currentContext?.size?.width;
```

Comment:
At this point you can be sure that `_textFieldKey.currentContext` is not null, so you can do `_textFieldKey.currentContext!`.